internal/api/lock.go

```diff
@@ -185,3 +185,38 @@ func lockInstallationDBMigrationOperation(c *Context, operationID string) (*mode
 		})
 	}
 }
+
+// lockDatabase synchronizes access to the given multitenant database across
+// potentially multiple provisioning servers.
+func lockDatabase(c *Context, databaseID string) (*model.MultitenantDatabase, int, func()) {
+	database, err := c.Store.GetMultitenantDatabase(databaseID)
+	if err != nil {
+		c.Logger.WithError(err).Error("failed to query database")
+		return nil, http.StatusInternalServerError, nil
+	}
+	if database == nil {
+		return nil, http.StatusNotFound, nil
+	}
+
+	locked, err := c.Store.LockMultitenantDatabase(databaseID, c.RequestID)
+	if err != nil {
+		c.Logger.WithError(err).Error("failed to lock database")
+		return nil, http.StatusInternalServerError, nil
+	} else if !locked {
+		c.Logger.Error("failed to acquire lock for database")
+		return nil, http.StatusConflict, nil
+	}
+
+	unlockOnce := sync.Once{}
+
+	return database, 0, func() {
```

Comment:
Is there a reason that when we use this pattern we don't use HTTP StatusOK for the middle return code? I know that `0` is "OK" according to UNIX conventions, but the other values in this method returned as the second return value are HTTP codes -- shouldn't we stick to that pattern?

I saw another function in another PR somewhat recently with this mix of conventions and I called it out, but I don't remember what the outcome of that was.

Reply:
Yeah, I stuck with `0` as that is how all the other lock helpers are set up as well. Totally okay with updating them all to be something different in the future, but just want to keep the standard till then.